kompendium-core/src/main/kotlin/org/leafygreens/kompendium/Kompendium.kt

```diff
@@ -44,6 +45,8 @@ object Kompendium {
     paths = mutableMapOf()
   )
 
+  var currentSecuritySchemeName: String? = null
```

Comment:
yes, I also don't like it. This is needed if we want to add the security scheme automatically for a given route path. If we still want to automatically assign schemes to routes (see other comment), a better solution might be to enable some bootstrapping so that KompendiumAuth could hook into the moment that a path item is added (something similar as you've done with calculatePath).